src/libsyntax/parse/mod.rs

```diff
@@ -306,128 +311,49 @@ pub fn stream_to_parser(sess: &ParseSess, stream: TokenStream) -> Parser<'_> {
     Parser::new(sess, stream, None, true, false)
 }
 
+fn shrink(span: Span, d_lo: u32, d_hi: u32) -> Span {
```

Comment:
```suggestion
fn shrink(span: Span, delta_lo: u32, delta_hi: u32) -> Span {
```

Reply:
Could this use `Range<u32>` instead? And behave more like "span slicing" than "span shrinking"?